Refactor out woke instructions (#158)
canonical/sphinx-docs-starter-pack: help-woke.rst

> :orphan:

Reviewer: Please don't use orphans, but include the file in the navigation. There's no reason to hide it.

Author: I can do this if you feel strongly about it but the resource is intended to be accessed in the context of the README. It's a sub-page.

Reviewer: Yes, it can be a subpage to the readme. But I feel pretty strongly about it that all pages should always be accessible from the navigation tree - otherwise it's very easy to miss that there is another page with information.

===================================
Inclusive language check exemptions
===================================

This page provides an overview of two inclusive language check exemption
methods for files written in reST format. See the `woke documentation`_ for
full coverage.

Exempt a word
-------------

To exempt an individual word, place a comment on a line immediately preceding

> the line containing the word in question. This special comment is to include

Reviewer: ```suggestion
the line containing the word in question. This special comment must include
```
Easier to understand

.. code-block:: none

   .. wokeignore:rule=whitelist
   This is your text. The word in question is here: whitelist. More text.

Here is an example of an exemption that acts upon an element (the string

> wokeignore:rule=master

Reviewer: This won't work - it'll leave a `wokeignore:rule=master` both in the GitHub rendering and in the HTML output generated from RST.
I just tested this again... RST support for wokeignore is horrible. It doesn't work as described in line 18.
Let me figure something out ...

Here is an example of an exemption that acts upon an element of a URL that is

> expressed using the link definition method (typically at the bottom of a file):

Reviewer: I don't find that clearer - will everyone who uses the starter pack know what the "link definition method" is?

The issue that we see a lot is that a URL contains "master", and that this is hard to fix with a comment in the line above (especially in MyST, btw). So what we should say is that for links, the best solution is to define them in a separate file or at the bottom of the page and put the comment in the line above the link definition.

Author: I will propose something.

>    .. wokeignore:rule=whitelist
>    This is your text. The word in question is here: whitelist. More text.
> 
> Here is an example of an exemption that acts upon an element (the string
> wokeignore:rule=master
> "master") of a URL. It is recommended to do this by using the standard reST
> method of placing the link at the bottom of the page (or in a separate file):

Reviewer: ```suggestion
   This is your text. The word in question is here: whitelist. More text. :none:`wokeignore:rule=whitelist,`

Here is an example of an exemption that acts upon an element (the string
"master") of a URL. It is recommended to do this by using the standard reST :none:`wokeignore:rule=master,`
method of placing the link at the bottom of the page (or in a separate file):
```

.. code-block:: none

   .. LINKS
   .. wokeignore:rule=master
   .. _link definition: https://some-external-site.io/master/some-page.html

Exempt an entire file
---------------------

A more drastic solution is to make an exemption for the contents of an entire
file.

> Start by placing file ``.wokeignore`` into your project's root directory. Then,

Reviewer: The starter pack already has this file, so you only need to edit it.

methods for files written in reST format. See the `woke documentation`_ for
full coverage.

Exempt a word
-------------

To exempt an individual word, place a custom ``none`` role (defined in the
``canonical-sphinx-extensions`` Sphinx extension) anywhere on the line
containing the word in question. The role syntax is:

.. code-block:: none

   :none:`wokeignore:rule=<SOME_WORD>,`

For instance:

.. code-block:: none

   This is your text. The word in question is here: whitelist. More text. :none:`wokeignore:rule=whitelist,`

To exempt an element of a URL, it is recommended to use the standard reST
method of placing links at the bottom of the page (or in a separate file). In
this case, a comment line is placed immediately above the URL line. The comment
syntax is:

.. code-block:: none

   .. wokeignore:rule=<SOME_WORD>

Here is an example where a URL element contains the string "master": :none:`wokeignore:rule=master,`

.. code-block:: none

   .. LINKS
   .. wokeignore:rule=master
   .. _link definition: https://some-external-site.io/master/some-page.html

You can now refer to the label ``link definition_`` in the body of the text.

> Here :none:`wokeignore:rule=master,` is an example where a URL element contains the string "master":

Reviewer: Duplicate of line 35.
